### PR TITLE
Add subscription test

### PR DESCRIPTION
**Pull request by orenyodfat**
For now , a very simple event (Reputation Mint) subscription via web socket as a POC.
Using apollo client lib https://github.com/apollographql

### CODE REVIEW

File: test/integration/Subscription.spec.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+          reputationMints {
+            contract
+            amount
+            address
+          }
+        }
+        `;
+
+      const subscriptionClient =await  createSubscriptionObservable(
+          'http://graph-node:8001/by-name/daostack', // GraphQL endpoint
```
Comment (tsuberim) [inline test/integration/Subscription.spec.ts]: should be `process.env.node_ws`

File: test/integration/Subscription.spec.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          SUBSCRIBE_QUERY,                                       // Subscription query
+          {address: accounts[0].address.toLowerCase()}                                                // Query variables
+        );
+
+      let event;
+
+      var consumer =await subscriptionClient.subscribe(eventData => {
```
Comment (tsuberim) [inline test/integration/Subscription.spec.ts]: I think its easier to do `await subscriptionClient.toPromise()` which just returns the first event as a promise.

File: test/integration/Subscription.spec.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            amount
+            address
+          }
+        }
+        `;
+
+      const subscriptionClient =await  createSubscriptionObservable(
+          'http://graph-node:8001/by-name/daostack', // GraphQL endpoint
+          SUBSCRIBE_QUERY,                                       // Subscription query
+          {address: accounts[0].address.toLowerCase()}                                                // Query variables
```
Comment (tsuberim) [inline test/integration/Subscription.spec.ts]: Where is this used inside the query?

File: test/integration/Subscription.spec.ts (hunk 2 above)
Comment (orenyodfat) [reply test/integration/Subscription.spec.ts]: Not sure how

Comment (tsuberim) [reply test/integration/Subscription.spec.ts]: you just call `toPromise` on the observable.

File: test/integration/util.ts [hunk 4]
```diff
@@ -60,3 +66,16 @@ export function padZeros(str: string, max = 36) {
     str = str.toString();
     return str.length < max ? padZeros("0" + str, max) : str;
 }
+
+
+const getWsClient = function(wsurl) {
```
Comment (tsuberim) [inline test/integration/util.ts]: maybe make these a single function `watchQuery(query: string): Observable` ?

File: tslint.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+    "defaultSeverity": "error",
+    "extends": [
+        "tslint:recommended"
+    ],
+    "jsRules": {},
+    "rules": {},
```
Comment (tsuberim) [inline tslint.json]: I like single quotes

**Comment by orenyodfat** [comment]
"Should we add such tests for all contracts? seems like duplicated work."

@tsuberim the test is just to make sure the subscription is working.not sure we should do that for all mappings.

File: tslint.json (hunk 5 above)
Comment (orenyodfat) [reply tslint.json]: This file is autogenerated by tslint init.
it also lint the code to use double quotes.
I do not care ..

Comment (tsuberim) [reply tslint.json]:
```
"rules": {
  "quotemark": [true, "single"]
}
```

**Comment by tsuberim** [comment]
@orenyodfat So is this sufficient? Also does the event fire even when a property is changed in an already existent entity?

File: tslint.json (hunk 5 above)
Comment (orenyodfat) [reply tslint.json]: lets argue about that :)

Comment (tsuberim) [reply tslint.json]: single quotes don't required pressing `shift`. Boom!

File: tslint.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    "defaultSeverity": "error",
+    "extends": [
+        "tslint:recommended"
+    ],
+    "jsRules": {},
+    "rules": {
+              "object-literal-sort-keys": false,
+              "only-arrow-functions": false,
+              "no-var-requires": false,
+              "prefer-const": false
```
Comment (tsuberim) [inline tslint.json]: single quote

**Pull request merged**